tests/functional/testAddons.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const assert = require('assert');
+const vpn = require('./helper.js');
+
+describe('Addons', function() {
+  this.ctx.authenticationNeeded = true;
+
+  it('Addons are loaded', async () => {
+    if (!(await vpn.isFeatureFlippedOff('addonSignature'))) {
+      await vpn.flipFeatureOff('addonSignature');
+      assert(await vpn.isFeatureFlippedOff('addonSignature'));
+    }
+
+    await vpn.setSetting('addon/customServer', 'false');
+
+    await vpn.waitForElementProperty(
+        'VPNAddonManager', 'loadCompleted', 'true');
+
+    let count = 0;
+    await vpn.waitForCondition(async () => {
+      count = parseInt(
+          await vpn.getElementProperty('VPNAddonManager', 'count'), 10);
+      return count > 0;
+    });
+
+    await vpn.setSetting('addon/customServer', 'true');
+    await vpn.setSetting(
+        'addon/customServerAddress',
+        'https://bakulf.github.io/vpn-addons-test/empty_manifest/');
+    await vpn.waitForCondition(async () => {
+      return parseInt(
+                 await vpn.getElementProperty('VPNAddonManager', 'count'),
+                 10) === 0;
+    });
+
+    await vpn.setSetting('addon/customServer', 'false');
+    await vpn.waitForCondition(async () => {
+      return parseInt(
+                 await vpn.getElementProperty('VPNAddonManager', 'count'),
+                 10) === count;
+    });
+
+    await vpn.setSetting('addon/customServer', 'true');
+    await vpn.setSetting(
+        'addon/customServerAddress',
+        'https://archive.mozilla.org/pub/vpn/addons/releases/latest/');
```

Comment:
Wouldn't a reset do this?

Reply:
yes but it does much more: logout the user, etc...

Reply:
tests do not run on the same machine and they do not share the same configuration. So, no. All should be fine.

Reply:
Well, in that case we probably don't even to do all of these resetting operations? Your call, no strong opinion.